### PR TITLE
Initial version

### PR DESCRIPTION
**Pull request by msrb**

### CODE REVIEW

File: victimsdb_lib/database.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+                        records[ecosystem].add(record)
+                    except ParseError as e:
+                        _logger.warning(str(e))
+        return cls(records=records)
+
+    @classmethod
+    def from_git_url(cls, git_url):
+        """Build database from GIT URL."""
+        with tempfile.TemporaryDirectory() as temp_dir:
+            subprocess.check_call(['git', 'clone', '--single-branch', git_url, temp_dir])
```
Comment (tisnik) [inline victimsdb_lib/database.py]: `--depth=1` is another good option together with `--single-branch`

File: victimsdb_lib/database.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+            db_dir = os.path.join(temp_dir, 'database')
+            return cls.from_dir(db_dir=db_dir)
+
+    def __contains__(self, cve_id):
+        if not cve_id:
```
Comment (tisnik) [inline victimsdb_lib/database.py]: nitpick: `if cve_id is not None` might be better.

File: victimsdb_lib/database.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+            return False
+
+        for records in self._records.values():
+            for record in records:
+                if record.cve_id == cve_id:
+                    return True
+        return False
+
+    def __getitem__(self, cve_id):
+        for records in self._records.values():
```
Comment (tisnik) [inline victimsdb_lib/database.py]: check for none?

File: victimsdb_lib/model.py [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+
+    def __init__(self, cve_id, title, description, cvss_v2, references, affected):
+        if cve_id:
+            cve_id = cve_id.strip()
+            if not cve_id.startswith('CVE-'):
+                cve_id = 'CVE-{cve_id}'.format(cve_id=cve_id)
+        self.cve_id = cve_id
+        self.title = title or ''
+        self.description = description or ''
+        self.cvss_v2 = cvss_v2 or None
```
Comment (tisnik) [inline victimsdb_lib/model.py]: tbh I don't understand why `or None`, but well why not, you are explicit here, so ok :)

File: victimsdb_lib/model.py [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+
+
+class VersionRange(object):
+    """Representation of a version range."""
+
+    def __init__(self, version_str):
+        # https://github.com/victims/victims-cve-db#version-string-common
+        self._version_str = version_str.strip()
+
+        result = re.fullmatch(r'^[><=]{1}=[^, ]+(,[^, ]+)?$', version_str.strip())
```
Comment (tisnik) [inline victimsdb_lib/model.py]: can't understand it right now (it's 23:22) :D

File: victimsdb_lib/database.py (hunk 1 above)
Comment (msrb) [reply victimsdb_lib/database.py]: Ah, right! Thanks. I knew there was such option, but from some reason I thought it was `--shallow` or something like that. Which didn't work.

File: victimsdb_lib/database.py (hunk 2 above)
Comment (msrb) [reply victimsdb_lib/database.py]: There should be no record in the database with an empty cve_id field. So no reason to continue, in my opinion. Although I am not sure now if I am enforcing it. In any case, the comment about no-empty-cve_id-records holds true so I'd keep it this way.

**Pull request merged**